draft-ietf-privacypass-architecture.md

```diff
@@ -425,7 +425,54 @@ their position to try and reduce the anonymity sets that Clients belong
 to (or, user segregation). For each case, we provide mitigations that
 the Privacy Pass ecosystem must implement to prevent these actions.
 
-## Metadata Privacy Implications
+## Correlation attacks
+
+One of the requirements of the issuance protocol as described in {{issuance-protocol}}
+is that it provide unconditional input secrecy. This means that an issuance
+and its corresponding redemption events are unlinkable using the contents of
+a token. However, there are other ways in which these two events could be linked
+together, which we refer to as correlation.
+
+Correlation can be problematic from a privacy perspective. To demonstrate why,
+consider the lifecycle of a cross-origin token. Clients may invoke the issuance
+protocol in one context, e.g., when visiting site foo.example, but redeem the
+token in another context, e.g., when visiting site bar.example. If an attacker
+can correlate these two events, then they know that visits to foo.example and
+bar.example are related. In contrast, per-origin token contexts are the same,
+so correlating issuance and redemption only allows an attacker to learn when
+issuance and redemption correspond to the same context.
+
+In general, each issuance and redemption event is associated with some Client
+and context. For example, an issuance event might represent Client C running
+the issuance protocol to produce a token for website foo.example at time T.
+Let cI and sI be the Client and context pair associated with an issuance event.
+Similarly, let cR and sR be the Client and context pair associated with a
+redemption event. We refer to context correlation as the case where cI = cR,
+and source correlation as the case where sI = sR.
+
+Context correlation is possible through a number of different ways. The following
+list covers some examples:
+
+- The timestamps of issuance and redemption events for interactive tokens may be
+  sufficiently unique that an attacker can link the two by their timestamp. However,
+  even if these timestamps are not close, a small number of issuance and redemption
+  events can let an attacker link two events. For example, if there is only ever
+  a single issuance and redemption event in a day, then an attacker can link based
+  based solely on the presence of the events.
+- The origin for an issuance and redemption event can be unique for a given Client,
+  allowing an attacker to link both events.
+
+Depending on the Client identifiers used, context correlation may not constitute
+a privacy violation. In particular, because tokens are not bound to a single Client
+and therefore transferrable, an attacker cannot distinguish the scenarios where
+the same Client ran the issuance protocol and redemption protocol, or where one
+Client ran the issuance protocol and another client ran the redemption protocol.
```

Comment:
I don't think this is in scope unless we are also going to discuss architectures and protocols for (safe?) client-to-client token sharing.

Reply:
This is noting that clients _can_ be transferred.

Reply:
I don't think we can hang our privacy safety claim on a client behavior that is never otherwise standardized, recommended, or even discussed.  Also, this kind of sharing is unsafe in general, and seems unlikely to occur in practice without substantial changes to Privacy Pass.

If you want to open this can of worms, I think you have to open it all the way.

Reply:
I think there's a misunderstanding here. Per the charter, a goal of this protocol is to "specify an extensible protocol for creating and redeeming anonymous and transferrable tokens." The key word is _transferrable_. This paragraph is simply acknowledging that tokens _can_ in fact be transferred amongst clients -- by design! It is by no means encouraging this behavior. The can of worms is already open.